### PR TITLE
feat: nonroot workspace

### PR DESCRIPTION
**Pull request by jjmaestro**
> [!NOTE]  
> Stacked on top of #39

Per the docs, when running with a nonroot user in the Docker image, we need to set the nonroot-workspace flag to True.

### CODE REVIEW

File: rules.bzl [hunk 1]
```diff
@@ -120,6 +124,7 @@ UBUNTU22_04_IMAGE = ":".join([UBUNTU22_04_REPOSITORY, "latest"])
 def buildbuddy(
         name,
         container_image = None,
+        nonroot_workspace = True,
```
Comment (jjmaestro) [inline rules.bzl:127]: I default to `True` because the docs say the Docker images are expected to run with the `buildbuddy` user... if this should default to `False` let me know!

Comment (bduffany) [reply rules.bzl:127]: Thanks for sending this change! I do think that we could benefit from simplifying the non-root setup. I think ideally though, we could reduce the need for some of this manual configuration. I will talk with the team internally about this, but I think we could either make all workspaces writable by non-root by default, or have nonroot-workspace default to true when dockerUser is a non-root user.

Comment (jjmaestro) [reply rules.bzl:127]: @bduffany agreed, having it "just work" would be even better than having the default in the toolchain. And piggybacking on `dockerUser` makes sense. But then I'd make it explicit in the toolchain because right now, it seems to be an implementation detail that's just "barely documented" in the docs... it took me a while to re-read it all and understand the user part + the nonroot part, etc.

I think the toolchain should set the `dockerUser` to `buildbuddy` by default and that would signal people using it that it could be changed to any other non-`root` (or `root`) user.

What do you think?

Comment (jjmaestro) [reply rules.bzl:127]: @bduffany / @sluongng ping! anything I should change in the PR? thanks!

Comment (sluongng) [reply rules.bzl:127]: I think it's a bit clunky to add this, especially as we are unclear on how nonroot is supported in different sandboxing options on our executor.

Instead of going down this route, what do you think about using platform inheritance https://bazel.build/reference/be/platforms-and-toolchains#platform_inheritance to define your platform that uses our platform as a parent?

```
platform(
    name = "my_platform",
    exec_properties = {
        "nonroot-workspace": "true",
    },
    parents = ["@buildbuddy_toolchain//:platform_linux_x86_64"],
)
```

Comment (jjmaestro) [reply rules.bzl:127]: @sluongng I see, I guess that would work. But I really don't see how this is "clunky"! 😅  Can I have some more context / explaining? The way I see it, your suggestion is "forcing users" to define their own `platforms` when the PR would enable custom non-root Docker images (like the ones BB is currently running, per the docs) to "just work". And it allows a simple config flag in the module extension that (also IMO) matches the BB documentation re. how to run with root/non-root Docker users.

So, TL;DR, I'm happy to get my stuff working with your suggestion but from my end-user perspective, I still think the PR merits discussion and potentially merging it / some form of it... at least from the time being now, until BB changes how it currently works.

Comment (sluongng) [reply rules.bzl:127]: Oh sorry, let me add some context/thoughts: I think this specific exec property would often couple with the `dockerUser` property, both implying a custom `container-image` being used. Right now we don't have a good way to recommend all of these exec properties that we support except for our doc + code. So adding them to this toolchain repo would create yet another place that we would have to update/sync whenever we change the default value.

Given that we are not certain what the default should be here just yet (pending internal discussion), I would rather not add this to the toolchain for the time being and direct folks to our doc website for accurate references (second only to our source code).

We should revisit how we should support all the exec properties listed in https://www.buildbuddy.io/docs/rbe-platforms in a more programmatic fashion in the future. Let me file an internal feature request for that.

Comment (jjmaestro) [reply rules.bzl:127]: Cool, understood. I still think having things "just work" would actually be better but I can also see the angle of "better leave things as-is" to avoid having potential constraints / stuff to maintain when you discuss it internally and decide which direction to take from here on.

I'll use the custom platform and close the PR then! Also, happy to contribute to the discussion and testing re. the future properties / config / whatever for this functionality :)

Comment (bduffany) [reply rules.bzl:127]: @jjmaestro We removed the `nonroot-workspace` property (making it a NOP). The change should go out in today's release :+1: Thanks again for bringing this to our attention!

Comment (jjmaestro) [reply rules.bzl:127]: @bduffany yay!! thanks a ton for moving so quick! :)
